pkg/rpm/rpm.go

```diff
@@ -16,6 +19,36 @@ func Version(name string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	out, err := exec.Command(rpmPath, options, name).Output()
-	return strings.TrimSpace(string(out)), err
+
+	// TODO: When we move to go 1.7+
+	//
+	//ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	//defer cancel()
+	//
+	//out, err := exec.CommandContext(ctx, rpmPath, options, name).Output()
+	//if ctx.Err() == context.DeadlineExceeded {
+	//	return nil, errors.New("rpm timed out")
+	//}
+	//return strings.TrimSpace(string(out)), err
+
+	// < go 1.7
+	cmd := exec.Command(rpmPath, options, name)
+
+	var out bytes.Buffer
+	cmd.Stdout = &out
+
+	cmd.Start()
+
+	done := make(chan error)
+	go func() { done <- cmd.Wait() }()
+
+	timeout := time.After(time.Minute)
```

Comment:
I think 1 minute is too long.  10 seconds would be a long time to wait.  Not sure if this should even be an error.Returning "timed out" in the version field would be fine, since this is only used by humans.  I have no idea how the rpm database got locked.